engine/crates/engine-v2/src/executor/mod.rs

```diff
@@ -0,0 +1,51 @@
+use std::sync::Arc;
+
+use engine_parser::types::OperationType;
+use futures_locks::Mutex;
+use schema::Resolver;
+
+use crate::{
+    response_graph::{Input, OutputNodeSelectionSet, ResponseGraph},
+    Engine,
+};
+
+mod coordinator;
+mod graphql;
+
+pub use coordinator::ExecutorCoordinator;
+
+struct ExecutorRequest<'a> {
+    operation_type: OperationType,
+    input: Input<'a>,
+    output: &'a OutputNodeSelectionSet,
+}
+
+enum Executor {
+    GraphQL(graphql::GraphqlExecutor),
+}
+
+impl Executor {
+    fn build(engine: &Engine, resolver: &schema::Resolver, request: ExecutorRequest<'_>) -> Self {
+        match resolver {
+            Resolver::Subgraph(resolver) => graphql::GraphqlExecutor::build(engine, resolver, request),
+        }
+    }
+
+    async fn execute(self, proxy: ResponseGraphProxy) {
+        match self {
+            Executor::GraphQL(plan) => plan.execute(proxy).await,
```

Comment:
Nitpicking , but the thing inside the `Executor::GraphQL` is a `GraphQlExecutor` - so `executor` seems like a less confusing name.

Reply:
yeah leftover